src/main/java/org/sonarlint/intellij/ui/WhatsInThisViewPanel.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2023 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.util.ui.JBUI
+import java.awt.CardLayout
+import java.awt.GridBagConstraints
+import java.awt.GridBagLayout
+import javax.swing.JLabel
+import javax.swing.JPanel
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.common.vcs.VcsListener
+import org.sonarlint.intellij.core.ProjectBindingManager
+import org.sonarlint.intellij.ui.UiUtils.Companion.runOnUiThread
+import org.sonarlint.intellij.util.HelpLabelUtils.Companion.createCurrentFileHelp
+import org.sonarlint.intellij.util.HelpLabelUtils.Companion.createCurrentFileHelpNotConnected
+import org.sonarlint.intellij.util.runOnPooledThread
+
+class WhatsInThisViewPanel(project: Project) {
+    private val NOT_CONNECTED = "Not Connected"
+    private val CONNECTED = "Connected"
+    private var project: Project? = null
+
+    var panel: JPanel? = null
+    private var layout: CardLayout? = null
+
+    init {
+        layout = CardLayout()
+        panel = JPanel(layout)
+        this.project = project
+        createPanel()
+        switchCards()
+        CurrentFileStatusPanel.subscribeToEventsThatAffectCurrentFile(project) { this.switchCards() }
+        project.messageBus.connect(project).subscribe<VcsListener>(
+            VcsListener.TOPIC,
+            VcsListener { _: Module?, _: String? ->
+                runOnUiThread(
+                    project
+                ) { this.switchCards() }
+            })
```

Comment:
You don't need this part here, as VCS events don't influence the content of the panel